quix-frontend/service/src/modules/event-sourcing/plugins/note-plugin.ts

```diff
@@ -59,76 +61,62 @@ export class NotePlugin implements EventBusPlugin {
-        let _model, _dbModel, _newModel;
-
-        try {
-          if (action.type === NoteActionTypes.addNote) {
-            const note = await noteReducer(undefined, action);
-            if (note) {
-              return this.noteRepository.insertNewWithRank(
-                convertNoteToDb(note),
-              );
-            }
-            return;
+        if (action.type === NoteActionTypes.addNote) {
+          const note = await noteReducer(undefined, action);
+          if (note) {
+            return this.noteRepository.insertNewWithRank(convertNoteToDb(note));
           }
-          const dbModel = await this.noteRepository.findOneOrFail(action.id);
-          _dbModel = dbModel;
+          return;
+        }
 
-          switch (action.type) {
-            case NoteActionTypes.reorderNote: {
-              return this.noteRepository.reorder(dbModel, action.to);
-            }
+        const dbModel = await this.noteRepository.findOneOrFail(action.id);
 
-            case NoteActionTypes.deleteNote: {
-              return this.noteRepository.deleteOneAndOrderRank(dbModel);
-            }
+        switch (action.type) {
+          case NoteActionTypes.reorderNote: {
+            return this.noteRepository.reorder(dbModel, action.to);
+          }
 
-            default: {
-              const model = convertDbNote(dbModel);
-              const newModel = noteReducer(model, action);
-              _model = model;
-              _newModel = newModel;
-              if (newModel && model !== newModel) {
-                return await this.noteRepository.save(
-                  convertNoteToDb(newModel),
-                  {
-                    reload: false,
-                  },
-                );
-              }
+          case NoteActionTypes.deleteNote: {
+            return this.noteRepository.deleteOneAndOrderRank(dbModel);
+          }
+          default: {
+            const model = convertDbNote(dbModel);
+            const newModel = noteReducer(model, action);
+
+            if (newModel && this.modelHasChanged(action, model, newModel)) {
+              return await this.noteRepository.save(convertNoteToDb(newModel), {
+                reload: false,
+              });
             }
           }
-        } catch (e) {
-          this.log(action, e, _model, _dbModel, _newModel);
-          throw e;
         }
       },
     );
   };
 
-  log = (
-    action: any,
-    e: any,
-    model: any = undefined,
-    dbModel: any = undefined,
-    newModel: any = undefined,
+  modelHasChanged = (
+    action: IAction<NoteActions>,
+    model: IBaseNote,
+    newModel: IBaseNote,
   ) => {
-    const msg = `
-    +===*Test*===+
-
-    Action: ${JSON.stringify(action)}
-    +========+
-    Error: ${e}
-    +========+
-    Model: ${JSON.stringify(model)}
-    +========+
-    DbModel: ${JSON.stringify(dbModel)}
-    +========+
-    NewModel: ${JSON.stringify(newModel)}
+    switch (action.type) {
```

Comment:
you need to handle the default case, assuming we are adding another note action, how it will be handled in general?

Reply:
done